trackintel/preprocessing/positionfixes.py

```diff
@@ -204,52 +204,51 @@ def generate_triplegs(positionfixes_in, spts=None, method='between_staypoints',
     if method == 'between_staypoints':
 
         # get case: # todo: @SVen I think only case 2 is now relevant
-        # Case 1: Staypoints are provided and are connected to positionfixes which have a column 'staypoint_id'
+        # Case 1: Positionfixes have a column 'staypoint_id'
         # Case 2: Staypoints are provided but positionfixes do not have a column 'staypoint_id'
-        # case 3: Staypoints are not provided but positionfixes have a column 'staypoint_id'
 
-        if spts is not None and "staypoint_id" in pfs:
-            case = 1
-        elif spts is not None:
+        if 'staypoint_id'not in pfs.columns:
             case = 2
-        elif "staypoint_id" in pfs:
-            case = 3
         else:
-            raise Exception('unknown case')
+            case = 1
 
-        # Preprocessing for case 2: Assign staypoint ids to positionfixes by matching timestamps
+        # Preprocessing for case 2:
+        # - step 1: Assign staypoint ids to positionfixes by matching timestamps (per user)
+        # - step 2: Find first positionfix after a staypoint (relevant if the positionfixes of staypoints are not
+        # provided)
         if case == 2:
             insert_index_psf_list = []
             pfs['staypoint_id'] = np.nan
             for user_id_this in pfs['user_id'].unique():
                 spts_u = spts[spts['user_id'] == user_id_this]
                 pfs_u = pfs[pfs['user_id'] == user_id_this]
 
-                # all positionfixes with timestamp between staypoints are assigned the value 0
-                # Here we intersect all positionfixes of a user with all staypoins of the same user
+                # step 1
+                # All positionfixes with timestamp between staypoints are assigned the value 0
+                # Intersect all positionfixes of a user with all staypoints of the same user
                 intervals = pd.IntervalIndex.from_arrays(spts_u['started_at'], spts_u['finished_at'], closed='both')
                 is_in_interval = pfs_u['tracked_at'].apply(lambda x: intervals.contains(x).any()).astype('bool')
                 pfs.loc[is_in_interval[is_in_interval].index, 'staypoint_id'] = 0
 
-                #
-                # find closest positionfix (gleich oder später)
+                # step 2
+                # Identify first positionfix after a staypoint
+                # find index of closest positionfix with equal or greater timestamp.
                 tracked_at_sorted = pfs_u['tracked_at'].sort_values()
                 insert_position_psf_u = tracked_at_sorted.searchsorted(spts_u['finished_at'])
                 insert_index_psf = tracked_at_sorted.iloc[insert_position_psf_u].index
 
                 # store the insert insert_position_spts_u in an array
-                insert_index_psf_list.extend(list(tracked_at_sorted.iloc[insert_position_psf_u].index))
+                insert_index_psf_list.extend(list(tracked_at_sorted.iloc[insert_index_psf].index))
```

Comment:
This causes a out of bound error when testing it with the geolife_long. removing the .iloc solves the error

Reply:
I think this is actually `insert_index_ls.extend(list(insert_index_psf))`